### PR TITLE
Update dependencies

### LINKED ISSUE #1609: Update dusk dependencies
##### Update all dusk dependencies versions:
| lib | from version | to version |
| --- | --- | --- |
| `dusk-bls12_381` | 0.12 | 0.13 |
| `dusk-jubjub` | 0.13 | 0.14 |
| `dusk-plonk` | 0.16 | 0.19 |
| `dusk-poseidon` | 0.31 | 0.33 |
| `poseidon-merkle` | 0.3 | 0.5 |
| `piecrust` | 0.17 | 0.18 |
| `phoenix-core` | 0.21 | 0.26 |
| `zk-citadel` | 0.5 | 0.10 |

##### rename lib:
| old name at version | new name at version |
| --- | --- |
| `bls12_381-sign` 0.6 | `bls12_381-bls` 0.2 |
| `dusk-schnorr` 0.18 | `jubjub-schnorr` 0.2 |

##### deprecate `dusk-pki` move objects to `jubjub-schnorr` and `phoenix-core`:
| `dusk-pki` struct or trait | moved to | will be used as |
| --- | --- | --- |
| `SecretKey` | `jubjub_schnorr::SecretKey` | `NoteSecretKey` |
| `PublicKey` | `jubjub_schnorr::PublicKey` | `NotePublicKey` |
| `SecretSpendKey` | `phoenix_core::SecretKey` | `SecretKey` |
| `PublicSpendKey` | `phoenix_core::PublicKey` | `PublicKey` |
| `StealthAddress` | `phoenix_core::StealthAddress` | - |
| `ViewKey` | `phoenix_core::ViewKey` | - |
| `Ownable` | `phoenix_core::Ownable` | - |

With the keys changing their names, this PR also updates function and variable names accordingly:

- `ssk` -> `sk`
- `psk` -> `pk`
- `SecretKey` and `PublicKey` from the `bls12_381-bls` lib will be imported as `StakeSecretKey` and `StakePublicKey` respectively
- wallet functions:
  - `wallet.public_spend_key` -> `wallet.public_key`
  - `wallet.public_key` -> `wallet.stake_public_key`
  - `retrieve/derive_sk` -> `retrieve/derive_stake_sk`
  - `retrieve/derive_ssk` -> `retrieve/derive_sk`

### PR DESCRIPTION
**Pull request by moCello**
Resolves #1609

### CODE REVIEW

**Comment by moCello** [comment]
closed by accident

**Review by ureeves — APPROVED** [review]
LGTM

File: node/benches/accept.rs
[the remark this answers is not in the document]
Comment (moCello) [reply node/benches/accept.rs:1]: They do!

File: contracts/license/tests/license.rs [hunk 1]
```diff
@@ -126,20 +127,21 @@ fn license_issue_get_merkle() {
     let mut session = initialize();
 
     // user
-    let ssk_user = SecretSpendKey::random(rng);
-    let psk_user = ssk_user.public_spend_key();
-    let sa_user = psk_user.gen_stealth_address(&JubJubScalar::random(rng));
+    let sk_user = SecretKey::random(rng);
+    let pk_user = PublicKey::from(&sk_user);
+    let sa_user = pk_user.gen_stealth_address(&JubJubScalar::random(&mut *rng));
```
[the remark this answers is not in the document]
Comment (moCello) [reply contracts/license/tests/license.rs:132]: same answer 😅 when using `rng` the function consumes it

File: circuits/transfer/src/execute/output.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ impl CircuitOutput {
     /// Zeroed stealth address used for padding
     pub const ZERO_STEALTH_ADDRESS: StealthAddress = {
         let pk = Self::ZERO_COMMITMENT;
-        let pk = PublicKey::from_raw_unchecked(pk);
+        let pk = NotePublicKey::from_raw_unchecked(pk);
```
Comment (xevisalle) [inline circuits/transfer/src/execute/output.rs:35]: Same, use npk or note_pk here

**Review by xevisalle — CHANGES_REQUESTED** [review]
Very good! Just the comment on the note keys! I commented two, but there some other places with this notation in the code.

File: circuits/transfer/src/execute/output.rs (hunk 2 above)
Comment (moCello) [reply circuits/transfer/src/execute/output.rs:35]: I was going to change the variable names when also changing the method names, but I might as well do it now

Comment (xevisalle) [reply circuits/transfer/src/execute/output.rs:35]: As you prefer then! I approve.

**Review by xevisalle — APPROVED** [review]
LGTM, considering that the keys names will be changed in a future PR.

**Review by HDauven — APPROVED** [review]

**Pull request merged**